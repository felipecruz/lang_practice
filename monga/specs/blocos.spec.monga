void func () {
    int x;
    {
        a[10] = 1;
        {
            int[] a;
            int[] a;
            int[] a;
            return;
        }
    }
}

int func () {
    int[] a;
    a[0] = 1;
    return a;
}
