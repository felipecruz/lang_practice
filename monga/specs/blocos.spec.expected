
Func Decl - Return Type: Void 
          - Paramenters 
   Var Decl Int Name x 
    Assignment Var: a [EXP: Const int: 10]<- EXP: Const int: 1
     Var Decl Int[] Name a 
     Var Decl Int[] Name a 
     Var Decl Int[] Name a 
     Return 

Func Decl - Return Type: Int 
          - Paramenters 
   Var Decl Int[] Name a 
   Assignment Var: a [EXP: Const int: 0]<- EXP: Const int: 1
   Return EXP: Var: a 

